sdk/protocol-plugins/tests/utils/ProtocolPluginMock.ts

```diff
@@ -102,14 +107,17 @@ export class ProtocolPluginMock implements IProtocolPlugin {
     return undefined as unknown as IPosition
   }
 
-  getActionBuilder<T extends steps.Steps>(step: T): Maybe<IActionBuilder<T>> {
-    const builder = this.stepBuilders[step.type]
+  getActionBuilder<
+    StepType extends SimulationSteps,
+    Step extends FilterStep<StepType, steps.Steps>,
+  >(stepType: StepType): Maybe<IActionBuilder<Step>> {
+    const builder = this.stepBuilders[stepType]
 
     if (!builder) {
       return undefined
     }
 
-    return new builder() as IActionBuilder<T>
+    return new builder() as IActionBuilder<Step>
```

Comment:
I saw that we are use casting to type it in other `getActionBuilder`s as well - is it necessary since function return is already typed with `Maybe<IActionBuilder<Step>>`?

Reply:
Indeed, I've fixed it. Good catch!